### PR TITLE
Correcting metric types for a few metrics

### PR DESCRIPTION
**Pull request by keitwb**
Also sending Garden app metadata as both properties and dimensions to satisfy
current nav view constraints.

Adding License file as well.

### CODE REVIEW

File: metrics/firehose_nozzle.go [hunk 1]
```diff
@@ -161,10 +163,16 @@ func (o *SignalFxFirehoseNozzle) datapointsFromEnvelope(envelope *events.Envelop
         dimensions["app_instance_index"] = strconv.Itoa(int(contMetric.GetInstanceIndex()))
         dimensions["app_id"] = guid
 
+        // Send app metadata as both dims and properties since navigator views
+        // seem to really want them as properties.
         properties["app_name"] = o.metadataFetcher.GetAppNameForGUID(guid)
         properties["app_space"] = o.metadataFetcher.GetSpaceNameForGUID(guid)
         properties["app_org"] = o.metadataFetcher.GetOrgNameForGUID(guid)
 
+        dimensions["app_name"] = properties["app_name"]
```
Comment (jrcamp) [inline metrics/firehose_nozzle.go]: I don't understand why duplicate properties and dimensions. Seems like it should only be dimensions. A property is attached to a dimension.

File: metrics/metric_types.go [hunk 2]
```diff
@@ -1,30 +1,42 @@
...
+    },
 }
 
+//var metricTypeToHuman = map[datapoint.MetricType]string{
+    //datapoint.Counter: "cumulative_counter",
+    //datapoint.Gauge: "gauge",
+//}
+
 func datapointType(origin string, metricName string, defaultType datapoint.MetricType) datapoint.MetricType {
-	originMap := metricNameToType[origin]
```
Comment (jrcamp) [inline metrics/metric_types.go]: Should really set your editor to use tabs for *.go and run go fmt on everything.

File: metrics/firehose_nozzle.go (hunk 1 above)
Comment (keitwb) [inline metrics/firehose_nozzle.go]: When selecting what fields to show in charts (e.g. in list charts, which items to show as the item name) they have to be dimensions, properties don't show up.  In the navigator views, when selecting what to show in the list below the heatmap as well as what shows up in the tooltip for each host, they seem to have to be properties to work.  Dimensions don't work based on my testing.

If you know any other workarounds let me know.

File: metrics/metric_types.go (hunk 2 above)
Comment (keitwb) [inline metrics/metric_types.go]: Probably after this commit I will reformat everything with tabs to be proper style.  Until then I want it all to at least be consistent with the spaces.

Comment (charliesignalfx) [inline metrics/metric_types.go]: fwiw https://golang.org/cmd/gofmt/ gofmt can be configured to run on save in vscode.  If you're using that.

**Pull request merged**